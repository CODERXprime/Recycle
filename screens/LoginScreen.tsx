import React from 'react';
import SignUp from '../components/SignUp'

export default class LoginScreen extends React.Component {
  static navigationOptions = {
    title: 'MyProfile',
    headerStyle: {
      backgroundColor: '#518e30',
      marginBottom: 0,
    },
    headerTintColor: "white",
  };

  render() {
    return <SignUp />;
  };
};
