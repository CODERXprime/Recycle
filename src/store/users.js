import axios from 'axios';

const userReducer = (state = [], action) => {
  return state;
};

export default userReducer;
