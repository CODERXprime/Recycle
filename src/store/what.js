/*
FETCHING DATA FROM CLOUD VISION
GET http://
*/

import axios from 'axios';

const whatReducer = (state = [], action) => {
  return state;
};

export default whatReducer;
