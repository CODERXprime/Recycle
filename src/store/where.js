/*
FETCHING DATA FROM EARTH911 API in JSON
GET http://api.earth911.com/earth911.methodName
http://api.earth911.com/
*/

import axios from 'axios';

const FIND_PLACES_TO_RECYCLE = 'FIND_PLACES_TO_RECYCLE';

const _findPlacesToRecycle = locationDetails => ({ type: FIND_PLACES_TO_RECYCLE, locationDetails });

const initialState = [];

const whereReducer = (state = initialState, action) => {
  switch (action.type) {
    case FIND_PLACES_TO_RECYCLE:
      return action.locationDetails;
    default: return state;
  };
};

export const findPlacesToRecycle = (api_key, GPS, productInfo) => dispatch => {
  console.log(productInfo)
  /*
  1) GET MATERIAL ID: materials = api.earth911.searchMaterials({
    'api_key': api_key,
    'query': 'used motor oil',
})
  2) {We may be able to skip this call with GPS coordinates from cell phone}
  Convert Zip Code to Lat/Long: postal_data = api.earth911.getPostalData({
    'api_key': api_key,
    'country': 'US',
    'postal_code': '90210',
})
  3) With Lat/Long, can now find what locations accept certain recycables
    results = api.earth911.searchLocations({
   'api_key': api_key,
   'latitude': 34.09,
   'longitude': -118.41,
   'material_id': [1, 4],
   'max_distance': 5,
})

len(results) >>> 57

RES:
{'curbside': False,
 'description': 'Jim Falk Lexus of Beverly Hills',
 'distance': 1.7,
 'latitude': 34.07,
 'location_id': '2VwAkLGZ',
 'location_type_id': 0,
 'longitude': -118.39,
 'municipal': False}

  4) Expand on location details
  details = api.earth911.getLocationDetails({
  'api_key': api_key,
  'location_id': '2VwAkLGZ',
})

RES:
{'2VwAkLGZ': {'address': '9230 Wilshire Boulevard',
              'city': 'BEVERLY HILLS',
              'country': 'US',
              'created': '2007-12-18T16:03:00',
              'curbside': False,
              'description': 'Jim Falk Lexus of Beverly Hills',
              'event_only': False,
              'fax': '',
              'flag': 'none',
              'geocoded': True,
              'hours': 'Please call for hours of operation.',
              'latitude': 34.067096470199999,
              'location_type_id': 0,
              'longitude': -118.39313399700001,
              'materials': [{'business': False,
                             'business_method': 'none',
                             'description': 'Used Motor Oil',
                             'dropoff': True,
                             'material_id': 1,
                             'notes': '',
                             'pickup': False,
                             'residential': True,
                             'residential_method': 'dropoff'}],
              'municipal': False,
              'national': False,
              'notes': '',
              'phone': '(310) 274-5200',
              'postal_code': '90212',
              'province': 'CA',
              'region': 'LOS ANGELES',
              'updated': '2007-12-21T11:23:00',
              'url': ''}}
  */

  materials = api.earth911.searchMaterials({
    //this will have to talk to './what.js' in that the query is populated by the vision component.
    api_key,
    'query': 'used motor oil',
  });


  results = api.earth911.searchLocations({
    api_key,
    'latitude': GPS.latitude,
    'longitude': GPS.longitude,
    //get lat and long from phone GPS
    'material_id': [1, 4],
    //params below are optional
    'max_distance': 5,
    'max_results': 10,
  });

  locationDetails = api.earth911.getLocationDetails({
    api_key,
    'location_id': '2VwAkLGZ',
    //send this back to RecPlacesCard.js
  });

  //probably break this out into multiple thunks

  return axios
    //template until we get API KEY
    .get(`http://api.earth911.com/earth911.methodName?key=${api_key}&productInfo=${productInfo}`)
    .then(res => res.data)
    .then(locationDetails => dispatch(_findPlacesToRecycle(locationDetails)));
};

export default whereReducer;
