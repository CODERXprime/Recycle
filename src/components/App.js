import React, { Component } from 'react';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import { connect } from 'react-redux';

import { api_key } from '../../env';
import { findPlacesToRecycle } from '../store/where';
import RecPlacesCard from './RecPlacesCard';

const GPS = {
  latitude: '',
  longitutde: '',
}

class App extends Component {

  render() {
    return (
      <Router>
        <div />
      </Router>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  findPlacesToRecycle: (api_key, GPS, productInfo) => dispatch(findPlacesToRecycle(api_key, GPS, productInfo))
});

export default connect(null, mapDispatchToProps)(App);
