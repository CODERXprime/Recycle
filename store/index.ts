import { createStore, applyMiddleware, combineReducers } from 'redux';
import logger from 'redux-logger';
import thunk from 'redux-thunk';

import what from './what';
import where from './where';
import users from './users';
import materials from './materials';
import toggle from './toggle';

export interface reducerLayout {
  state: string[],
  action: object | null,
};

const reducer = combineReducers({
  what,
  where,
  users,
  materials,
  toggle,
});

export default createStore(reducer, applyMiddleware(logger, thunk));
