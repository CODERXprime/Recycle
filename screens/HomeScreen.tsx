import * as React from 'react';
import {
  Button,
  Image,
  Platform,
  ScrollView,
  StyleSheet,
  Text,
  TouchableOpacity,
  View,
} from 'react-native';
import { WebBrowser } from 'expo';

import { MonoText } from '../components/StyledText';
import Outcome from '../components/Outcome';
import CameraComp from '../components/CameraComp';

let proTips = ['Why did a chicken carrying a tin can cross the road? The recycling bin was on the other side.',
      'Scientists discovered a way to generate light using only scrap metal. It was an aluminating experience.',
      'What do you get when you have a glass bottle, some tin foil, and an old notebook? Recycling.',
      'Flip it over! Check it out before you chuck it out!',
      'Save the Earth! Its the only planet with Chocolate. So far…. (edited)',
      'Don’t be trashy! Recycle!', 'It’s easy being green- Reduce, Reuse, Recycle.',
      'Recycling plastic feels fantastic!', 'You will produce about 127, 604 pounds of garbage in your lifetime. Recycle.',
      'Have you hugged your recycling bin today?']

const randoNum = () => {
  return Math.floor((Math.random() * 10) + 1);
}

let num = randoNum();

const randomProTip = proTips[num];



export default class HomeScreen extends React.Component {
  static navigationOptions = {
    title: 'Recycle It!',
    headerStyle: {
      backgroundColor: '#518e30',
      marginBottom: 0,
    },
    headerTintColor: "white",
    headerTitleStyle: {
      fontSize: 35,
    },
  };

  render() {
    return (
      <View style={styles.container}>
        <ScrollView style={styles.container} contentContainerStyle={styles.contentContainer}>
          <View>
            <CameraComp />
          </View>
          <View>
            <Outcome />
          </View>
        </ScrollView>
        <View style={styles.tabBarInfoContainer}>
          <Text style={styles.tabBarInfoText}>{randomProTip}</Text>
        </View>
      </View>
    );
  };
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    margin: 0,
    padding: 0,
  },
  contentContainer: {
    paddingTop: 0,
  },
  welcomeImage: {
    width: 100,
    height: 80,
    resizeMode: 'contain',
    marginTop: 3,
    marginLeft: -10,
  },
  getStartedContainer: {
    alignItems: 'center',
    marginHorizontal: 50,
  },
  homeScreenFilename: {
    marginVertical: 7,
  },
  getStartedText: {
    fontSize: 17,
    color: 'rgba(96,100,109, 1)',
    lineHeight: 24,
    textAlign: 'center',
  },
  tabBarInfoContainer: {
    position: 'absolute',
    bottom: 0,
    left: 0,
    right: 0,
    ...Platform.select({
      ios: {
        shadowColor: 'black',
        shadowOffset: { height: -3 },
        shadowOpacity: 0.1,
        shadowRadius: 3,
      },
      android: {
        elevation: 20,
      },
    }),
    alignItems: 'center',
    backgroundColor: '#8e3051',
    paddingVertical: 10,
  },
  tabBarInfoText: {
    fontSize: 12,
    color: 'white',
    textAlign: 'center',
  },
  helpContainer: {
    alignItems: 'center',
  },
  helpLink: {
    paddingVertical: 15,
  },
  helpLinkText: {
    fontSize: 14,
    color: '#2e78b7',
  },
});
