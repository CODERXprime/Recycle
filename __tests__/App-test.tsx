import 'react-native';
import * as React from 'react';
import App from '../App';
import { Provider } from 'react-redux';
import store from '../store/index';
import HomeScreen from '../screens/HomeScreen';
import LoginScreen from '../screens/LoginScreen';
import * as renderer from 'react-test-renderer';
import NavigationTestUtils from 'react-navigation/NavigationTestUtils';


describe('App snapshot', () => {
  jest.useFakeTimers();
  beforeEach(() => {
    NavigationTestUtils.resetInternalState();
  });

  it('renders the loading screen', async () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the root without loading screen', async () => {
    const tree = renderer.create(<App skipLoadingScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders the home screen', async () => {
    const tree = renderer.create(
      <Provider store={ store }>
        <HomeScreen />
      </Provider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders the map screen', async () => {
    const tree = renderer.create(
      <Provider store={ store }>
        <LoginScreen />
      </Provider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  })
});
