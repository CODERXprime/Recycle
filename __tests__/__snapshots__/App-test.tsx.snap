// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot renders the home screen 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "margin": 0,
      "padding": 0,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#8e3051",
        "bottom": 0,
        "left": 0,
        "paddingVertical": 10,
        "position": "absolute",
        "right": 0,
        "shadowColor": "black",
        "shadowOffset": Object {
          "height": -3,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 3,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 12,
          "textAlign": "center",
        }
      }
    >
      Pro Tip #1: Recycling is good for you and the planet!
    </Text>
  </View>
</View>
`;

exports[`App snapshot renders the loading screen 1`] = `null`;

exports[`App snapshot renders the map screen 1`] = `
<View>
  <View>
    <View
      style={
        Array [
          Object {},
          undefined,
        ]
      }
    >
      <Text
        style={
          Array [
            Object {},
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Array [
              Object {
                "color": "#86939e",
                "fontSize": 15,
                "fontWeight": "bold",
                "marginBottom": 1,
                "marginLeft": 20,
                "marginRight": 20,
                "marginTop": 15,
              },
              undefined,
              undefined,
            ],
          ]
        }
      >
        User Name
      </Text>
    </View>
    <View
      style={
        Object {
          "borderBottomColor": "#bdc6cf",
          "borderBottomWidth": 1,
          "marginLeft": 20,
          "marginRight": 20,
          "transform": Array [
            Object {
              "translateX": 0,
            },
          ],
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        style={
          Array [
            Object {
              "color": "#86939e",
              "minHeight": 36,
              "width": 750,
            },
            Object {
              "fontSize": 17.5,
            },
            undefined,
          ]
        }
        underlineColorAndroid="transparent"
      />
    </View>
    <View
      style={
        Array [
          Object {},
          undefined,
        ]
      }
    >
      <Text
        style={
          Array [
            Object {},
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Array [
              Object {
                "color": "#86939e",
                "fontSize": 15,
                "fontWeight": "bold",
                "marginBottom": 1,
                "marginLeft": 20,
                "marginRight": 20,
                "marginTop": 15,
              },
              undefined,
              undefined,
            ],
          ]
        }
      >
        Email
      </Text>
    </View>
    <View
      style={
        Object {
          "borderBottomColor": "#bdc6cf",
          "borderBottomWidth": 1,
          "marginLeft": 20,
          "marginRight": 20,
          "transform": Array [
            Object {
              "translateX": 0,
            },
          ],
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        style={
          Array [
            Object {
              "color": "#86939e",
              "minHeight": 36,
              "width": 750,
            },
            Object {
              "fontSize": 17.5,
            },
            undefined,
          ]
        }
        underlineColorAndroid="transparent"
      />
    </View>
    <View
      style={
        Array [
          Object {},
          undefined,
        ]
      }
    >
      <Text
        style={
          Array [
            Object {},
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Array [
              Object {
                "color": "#86939e",
                "fontSize": 15,
                "fontWeight": "bold",
                "marginBottom": 1,
                "marginLeft": 20,
                "marginRight": 20,
                "marginTop": 15,
              },
              undefined,
              undefined,
            ],
          ]
        }
      >
        Password
      </Text>
    </View>
    <View
      style={
        Object {
          "borderBottomColor": "#bdc6cf",
          "borderBottomWidth": 1,
          "marginLeft": 20,
          "marginRight": 20,
          "transform": Array [
            Object {
              "translateX": 0,
            },
          ],
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        secureTextEntry={true}
        style={
          Array [
            Object {
              "color": "#86939e",
              "minHeight": 36,
              "width": 750,
            },
            Object {
              "fontSize": 17.5,
            },
            undefined,
          ]
        }
        underlineColorAndroid="transparent"
      />
    </View>
    <View
      style={
        Array [
          Object {},
          undefined,
        ]
      }
    >
      <Text
        style={
          Array [
            Object {},
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Array [
              Object {
                "color": "#86939e",
                "fontSize": 15,
                "fontWeight": "bold",
                "marginBottom": 1,
                "marginLeft": 20,
                "marginRight": 20,
                "marginTop": 15,
              },
              undefined,
              undefined,
            ],
          ]
        }
      />
    </View>
    <View
      style={
        Array [
          Object {
            "marginLeft": 15,
            "marginRight": 15,
          },
          Object {
            "shadowColor": "rgba(0,0,0, .4)",
            "shadowOffset": Object {
              "height": 1,
              "width": 1,
            },
            "shadowOpacity": 1,
            "shadowRadius": 1,
          },
          undefined,
          undefined,
        ]
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={null}
      >
        <View
          pointerEvents="box-only"
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#9E9E9E",
                "flexDirection": "row",
                "justifyContent": "center",
                "padding": 19,
              },
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Object {
                "backgroundColor": "#3E9428",
              },
              undefined,
              Object {
                "padding": 12,
              },
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
            ]
          }
        >
          <Text
            style={
              Array [
                Object {},
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Array [
                  Object {
                    "color": "white",
                    "fontSize": 20,
                  },
                  undefined,
                  Object {
                    "fontSize": 17.5,
                  },
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                ],
              ]
            }
          >
            Sign Up
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "marginLeft": 15,
            "marginRight": 15,
          },
          Object {
            "shadowColor": "rgba(0,0,0, .4)",
            "shadowOffset": Object {
              "height": 1,
              "width": 1,
            },
            "shadowOpacity": 1,
            "shadowRadius": 1,
          },
          undefined,
          undefined,
        ]
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "marginTop": 10,
          }
        }
      >
        <View
          pointerEvents="box-only"
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#9E9E9E",
                "flexDirection": "row",
                "justifyContent": "center",
                "padding": 19,
              },
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Object {
                "backgroundColor": "#3E9428",
              },
              undefined,
              Object {
                "padding": 12,
              },
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
            ]
          }
        >
          <Text
            style={
              Array [
                Object {},
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Array [
                  Object {
                    "color": "white",
                    "fontSize": 20,
                  },
                  undefined,
                  Object {
                    "fontSize": 17.5,
                  },
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                ],
              ]
            }
          >
            Log In
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`App snapshot renders the root without loading screen 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "overflow": "hidden",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        collapsable={false}
        pointerEvents="auto"
        removeClippedSubviews={false}
        style={
          Array [
            Object {
              "flex": 1,
              "overflow": "hidden",
            },
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "opacity": 1,
              },
            ],
          ]
        }
      >
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View
            onLayout={[Function]}
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              onMoveShouldSetResponder={[Function]}
              onMoveShouldSetResponderCapture={[Function]}
              onResponderEnd={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderReject={[Function]}
              onResponderRelease={[Function]}
              onResponderStart={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              onStartShouldSetResponderCapture={[Function]}
              style={
                Array [
                  Object {
                    "flex": 1,
                    "flexDirection": "column-reverse",
                    "overflow": "hidden",
                  },
                  Object {
                    "backgroundColor": "#000",
                  },
                ]
              }
            >
              <View
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <View
                  accessibilityElementsHidden={false}
                  pointerEvents="auto"
                  style={
                    Object {
                      "backgroundColor": "#E9E9EF",
                      "bottom": 0,
                      "left": 0,
                      "opacity": 1,
                      "paddingTop": 64,
                      "position": "absolute",
                      "right": 0,
                      "shadowColor": "black",
                      "shadowOffset": Object {
                        "height": 0,
                        "width": 0,
                      },
                      "shadowOpacity": 0.2,
                      "shadowRadius": 5,
                      "top": 0,
                      "transform": Array [
                        Object {
                          "translateX": 0,
                        },
                        Object {
                          "translateY": 0,
                        },
                      ],
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "backgroundColor": "#fff",
                        "flex": 1,
                        "margin": 0,
                        "padding": 0,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "#8e3051",
                          "bottom": 0,
                          "left": 0,
                          "paddingVertical": 10,
                          "position": "absolute",
                          "right": 0,
                          "shadowColor": "black",
                          "shadowOffset": Object {
                            "height": -3,
                          },
                          "shadowOpacity": 0.1,
                          "shadowRadius": 3,
                        }
                      }
                    >
                      <Text
                        style={
                          Object {
                            "color": "white",
                            "fontSize": 12,
                            "textAlign": "center",
                          }
                        }
                      >
                        Pro Tip #1: Recycling is good for you and the planet!
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
              <View
                onLayout={[Function]}
                pointerEvents="box-none"
                style={
                  Object {
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "#518e30",
                      "transform": Array [
                        Object {
                          "translateX": 0,
                        },
                      ],
                    }
                  }
                >
                  <View
                    onLayout={[Function]}
                    pointerEvents="box-none"
                    style={
                      Object {
                        "backgroundColor": "#518e30",
                        "borderBottomColor": "#A7A7AA",
                        "borderBottomWidth": 0.5,
                        "height": 64,
                        "marginBottom": 0,
                        "paddingBottom": 0,
                        "paddingLeft": 0,
                        "paddingRight": 0,
                        "paddingTop": 20,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "bottom": 0,
                            "flexDirection": "row",
                            "left": 0,
                            "position": "absolute",
                            "right": 0,
                            "top": 0,
                          }
                        }
                      >
                        <View
                          pointerEvents="box-none"
                          style={
                            Object {
                              "alignItems": "center",
                              "backgroundColor": "transparent",
                              "bottom": 0,
                              "flexDirection": "row",
                              "justifyContent": "center",
                              "left": 0,
                              "opacity": 1,
                              "position": "absolute",
                              "right": 0,
                              "top": 0,
                            }
                          }
                        >
                          <Text
                            accessibilityTraits="header"
                            accessible={false}
                            allowFontScaling={true}
                            numberOfLines={1}
                            onLayout={[Function]}
                            style={
                              Object {
                                "color": "white",
                                "fontSize": 35,
                                "fontWeight": "600",
                                "marginHorizontal": 16,
                                "textAlign": "center",
                              }
                            }
                          >
                            Recycle It!
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      pointerEvents="box-none"
      style={
        Object {
          "backgroundColor": "#fff",
          "borderTopColor": "rgba(0, 0, 0, .3)",
          "borderTopWidth": 0.5,
          "flexDirection": "row",
          "height": 49,
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
        }
      }
    >
      <View
        accessibilityLabel="Home"
        accessible={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
            },
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "flexDirection": "column",
              "justifyContent": "flex-end",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "height": 29,
              },
              false,
              Object {
                "flex": 1,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "height": "100%",
                "justifyContent": "center",
                "minWidth": 25,
                "opacity": 1,
                "position": "absolute",
                "width": "100%",
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "#2f95dc",
                    "fontSize": 26,
                  },
                  Object {
                    "marginBottom": -3,
                  },
                  Object {
                    "fontFamily": "ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "height": "100%",
                "justifyContent": "center",
                "minWidth": 25,
                "opacity": 0,
                "position": "absolute",
                "width": "100%",
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "#ccc",
                    "fontSize": 26,
                  },
                  Object {
                    "marginBottom": -3,
                  },
                  Object {
                    "fontFamily": "ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
          </View>
        </View>
        <Text
          allowFontScaling={true}
          numberOfLines={1}
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "#3478f6",
              "fontSize": 11,
              "marginBottom": 1.5,
              "textAlign": "center",
            }
          }
        >
          Home
        </Text>
      </View>
      <View
        accessibilityLabel="Locations"
        accessible={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
            },
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "flexDirection": "column",
              "justifyContent": "flex-end",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "height": 29,
              },
              false,
              Object {
                "flex": 1,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "height": "100%",
                "justifyContent": "center",
                "minWidth": 25,
                "opacity": 0,
                "position": "absolute",
                "width": "100%",
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "#2f95dc",
                    "fontSize": 26,
                  },
                  Object {
                    "marginBottom": -3,
                  },
                  Object {
                    "fontFamily": "ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "height": "100%",
                "justifyContent": "center",
                "minWidth": 25,
                "opacity": 1,
                "position": "absolute",
                "width": "100%",
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "#ccc",
                    "fontSize": 26,
                  },
                  Object {
                    "marginBottom": -3,
                  },
                  Object {
                    "fontFamily": "ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
          </View>
        </View>
        <Text
          allowFontScaling={true}
          numberOfLines={1}
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "#929292",
              "fontSize": 11,
              "marginBottom": 1.5,
              "textAlign": "center",
            }
          }
        >
          Locations
        </Text>
      </View>
      <View
        accessibilityLabel="Map"
        accessible={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
            },
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "flexDirection": "column",
              "justifyContent": "flex-end",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "height": 29,
              },
              false,
              Object {
                "flex": 1,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "height": "100%",
                "justifyContent": "center",
                "minWidth": 25,
                "opacity": 0,
                "position": "absolute",
                "width": "100%",
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "#2f95dc",
                    "fontSize": 26,
                  },
                  Object {
                    "marginBottom": -3,
                  },
                  Object {
                    "fontFamily": "ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "height": "100%",
                "justifyContent": "center",
                "minWidth": 25,
                "opacity": 1,
                "position": "absolute",
                "width": "100%",
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "#ccc",
                    "fontSize": 26,
                  },
                  Object {
                    "marginBottom": -3,
                  },
                  Object {
                    "fontFamily": "ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
          </View>
        </View>
        <Text
          allowFontScaling={true}
          numberOfLines={1}
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "#929292",
              "fontSize": 11,
              "marginBottom": 1.5,
              "textAlign": "center",
            }
          }
        >
          Map
        </Text>
      </View>
      <View
        accessibilityLabel="My Profile"
        accessible={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
            },
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "flexDirection": "column",
              "justifyContent": "flex-end",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "height": 29,
              },
              false,
              Object {
                "flex": 1,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "height": "100%",
                "justifyContent": "center",
                "minWidth": 25,
                "opacity": 0,
                "position": "absolute",
                "width": "100%",
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "#2f95dc",
                    "fontSize": 26,
                  },
                  Object {
                    "marginBottom": -3,
                  },
                  Object {
                    "fontFamily": "ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "height": "100%",
                "justifyContent": "center",
                "minWidth": 25,
                "opacity": 1,
                "position": "absolute",
                "width": "100%",
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "#ccc",
                    "fontSize": 26,
                  },
                  Object {
                    "marginBottom": -3,
                  },
                  Object {
                    "fontFamily": "ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
          </View>
        </View>
        <Text
          allowFontScaling={true}
          numberOfLines={1}
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "#929292",
              "fontSize": 11,
              "marginBottom": 1.5,
              "textAlign": "center",
            }
          }
        >
          My Profile
        </Text>
      </View>
    </View>
  </View>
</View>
`;
